x-pack/plugins/enterprise_search/public/applications/esre/components/esre_guide/rank_aggregation_section.tsx

```diff
@@ -18,69 +18,80 @@ import { EsreGuideAccordion } from './esre_guide_accordion';
-                defaultMessage: 'Intelligently combines rankings without configuration',
-              }
-            )}
-          >
-            <RrfRankingPanel />
-          </EsreGuideAccordion>
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EsreGuideAccordion
-            id="linearCombinationAccordion"
-            icon={linearCombinationIllustration}
-            title={i18n.translate('xpack.enterpriseSearch.esre.linearCombinationAccordion.title', {
-              defaultMessage: 'Linear combination',
-            })}
-            description={i18n.translate(
-              'xpack.enterpriseSearch.esre.linearCombinationAccordion.description',
-              {
-                defaultMessage: 'Weighted results from multiple rankings',
-              }
-            )}
-          >
-            <LinearCombinationPanel />
-          </EsreGuideAccordion>
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    </EuiFlexItem>
-  </EuiFlexGroup>
-);
+export const RankAggregationSection: React.FC = () => {
+  const [currentExpanded, setCurrentExpanded] = useState<string | undefined>();
+
+  return (
+    <EuiFlexGroup alignItems="center">
+      <EuiFlexItem grow={4}>
+        <EuiFlexGroup direction="column" gutterSize="s" justifyContent="flexStart">
+          <EuiFlexItem grow={false}>
+            <EuiTitle>
+              <h2>
+                <FormattedMessage
+                  id="xpack.enterpriseSearch.esre.rankAggregationSection.title"
+                  defaultMessage="Use a rank aggregation method"
+                />
+              </h2>
+            </EuiTitle>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiText>
+              <p>
+                <FormattedMessage
+                  id="xpack.enterpriseSearch.esre.rankAggregationSection.description"
+                  defaultMessage="Optional methods for fusing or combining different rankings to achieve better overall ranking performance."
+                />
+              </p>
+            </EuiText>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem grow={6}>
+        <EuiFlexGroup direction="column">
+          <EuiFlexItem grow={false}>
+            <EsreGuideAccordion
+              id="rrfRankingAccordion"
+              icon={rrfRankingIllustration}
+              title={i18n.translate('xpack.enterpriseSearch.esre.rrfRankingAccordion.title', {
+                defaultMessage: 'RRF hybrid ranking',
+              })}
+              description={i18n.translate(
+                'xpack.enterpriseSearch.esre.rrfRankingAccordion.description',
+                {
+                  defaultMessage: 'Intelligently combines rankings without configuration',
+                }
+              )}
+              onToggle={(isOpen) => setCurrentExpanded(isOpen ? 'rrfRankingAccordion' : undefined)}
+              forceState={currentExpanded === 'rrfRankingAccordion' ? 'open' : 'closed'}
```

Comment:
I wonder if there's a way to extract some of this code into a function(s) without having to repeat it for every accordion. I haven't found a way to pass the `id` property that is defined at the same time as `onToggle` and `forceState`.

Reply:
Probably the best way to do this is to move this logic down into the `EsreGuideAccordion` component. You can turn `onToggle` in the `EsreGuideAccordion` into a function that takes an id as an argument, rather than one that takes a boolean as an argument. Then whether the accordion is open or closed is internal to EsreGuideAccordion, and the top-level state just passes an id and which id it wants to be open.

```
interface EsreGuideAccordionProps {
[...]
  id: string;
  onToggle?: (id: string | undefined) => void;
  openId?: string;
}

const EsreGuideAccordion: React.FC<EsreGuideAccordionProps> = ({onToggle, forceState}) => {

  return (<EuiAccordion
    forceState={ id === openId ? 'open' : 'closed'}
    onToggle={(isOpen: boolean) => onToggle(isOpen ? 'id' : undefined)}
  />);
};

const Parent = () => {
  const [openId, setOpenId] = useState<string | undefined>(undefined);
  return (
    <EsreGuideAccordion
       id="guideId"
       onToggle={setOpenId}
       openId={openId}
  />
}
```

Reply:
Thanks for the tip, I moved the logic into `EsreGuideAccordion`.